File: gr/strummable-guitar.hs
-- strummable guitar (jmcc) #11

import Sound.SC3 {- hsc3 -}

str :: UGen -> UGen -> UGen
str sc ix =
    let x = mouseX KR 0 1 Linear 0.2
        t = abs (hpz1 (x >* (0.25 + ix * 0.1)))
        e = decay t 0.05
        n = pinkNoise ix AR * e
        dt = 1 / midiCPS sc
        s = combL n dt dt 4
    in pan2 s (ix * 0.2 - 0.5) 1

gtr :: UGen
gtr =
    let scale = [52,57,62,67,71,76]
        strs = sum (zipWith str scale [0..])
    in leakDC (lpf strs 12000) 0.995

main :: IO ()
main = audition (out 0 gtr)

File: gr/strummable-guitar.fs
\ strummable guitar (jmcc) #11

: scale 52 57 62 67 71 76 ;

: str { sc ix }
    0 1 linear 0.2 MouseX.kr { x }
    x 0.25 ix 0.1 * + > HPZ1.kr abs { t }
    t 0.05 Decay.kr PinkNoise.ar * { n }
    sc MIDICPS Recip { dt }
    n dt dt 4 CombL.ar { s }
    s ix 0.2 * 0.5 - 1 Pan2.ar ;

: strs 52 0 str 57 1 str + 62 2 str + 67 3 str + 71 4 str + 76 5 str + ;

strs play
